### ISSUE TITLE
Docker no longer backgrounds, does not respect cmd+q for backgrounding

### ISSUE THREAD

**Issue by timfallmk** [docker/for-mac#6167]
The new version (`4.5.0 (74594)`) of Docker Desktop for macOS now boots into the foreground and cannot be backgrounded, either by use of cmd + q or with a menu item.

Previous versions of Docker Desktop were intended to be left running "backgrounded" by launching the app without the GUI frontend presentation, and maintaining access in the menu bar. The application would not show in the Dock, nor was the "Dashboard" UI on screen. If the GUI was opened (either by clicking the option, or using the menu bar icon) it could immediately be sent to the "background" by intercepting the standard quit key combo in macOS cmd + q.

This version (`4.5.0 (74594)`) seems to launch into the foreground, with the GUI open and the application in Dock. It also removes the ability to send the application to "background" via the key combo or the menu option. This leaves the application forcibly running in the foreground at all times. The only way to remove GUI elements is to quit Docker entirely.

![image](https://user-images.githubusercontent.com/5316117/153531962-930d501a-d045-4ed5-9fe5-e1ff89acc524.png)

This is probably just an oversight, but it radically alters the behavior of Docker Desktop and makes a jarring change for any user that is not prepared for it.

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 8AA04ED2-FE4E-4107-A277-45CCD6E4EF73/20220211030400

### Expected behavior
As before, Docker Desktop starts in "background" mode, with only the menu bar icon showing. It can be opened and then "backgrounded" again via a menu option or cmd + q.

### Actual behavior
Docker Desktop launches initially in the foreground and stays that way. The only way to remove it from foreground is the quit the app entirely.

### Information

  - macOS Version: 12.2
  - Intel chip or Apple chip: Apple Silicon
  - Docker Desktop Version: `4.5.0 (74594)`

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
Diagnostics uploaded

### Steps to reproduce the behavior

Install version 4.5.0 (74594), reopen.

**Comment by timfallmk** [comment]
Can we get some eyeballs on this? It seems to be affecting a lot of people.

**Comment by mdeliadi** [comment]
Yes please, it's a very annoying issue!

**Comment by stenmigerode** [comment]
I have the same on Mac OS X 12.2 on intel: shortcuts only work when I the Docker from the Mac OS X Menu bar.

**Comment by trungutt** [comment]
Could you please try these builds to see if it could improve the situation? @stenmigerode @mdeliadi @timfallmk 
Mac Intel: https://desktop-stage.docker.com/mac/main/amd64/75234/Docker.dmg
Mac M1: https://desktop-stage.docker.com/mac/main/arm64/75234/Docker.dmg
Windows: https://desktop-stage.docker.com/win/main/amd64/75234/Docker%20Desktop%20Installer.exe

**Comment by stenmigerode** [comment]
Hi @trungutt,

Thanks a lot, but for me it's not working on Mac OS X 12.2.1 - Intel

When I restart the laptop, Docker Desktop is still opening in the foreground (although no windows are visible). When activating the Docker Desktop that runs in the foreground by command tabbing, none of the shortcuts work. I need to click the Docker Desktop icon in the toolbar on the top right to show the quickmenu: only then I can restart Docker Desktop.

When restarting docker desktop, the dashboard opens up in the foreground and I can use the Command+W shortcut to hide the foreground application.

So for me the problem is present when starting or rebooting the computer. In previous versions (before 4.5), Docker Desktop started in the background.

Kind regards,
Sten

**Comment by mdeliadi** [comment]
@trungutt hello again, unfortunately after restarting all the issues pointed out by @stenmigerode araised!

**Comment by trungutt** [comment]
Thanks, that's very helpful. Please let me try again one more, could you try these builds to see if it could improve the situation? @stenmigerode @mdeliadi 

Mac Intel: https://desktop-stage.docker.com/mac/main/amd64/75249/Docker.dmg
Mac M1: https://desktop-stage.docker.com/mac/main/arm64/75249/Docker.dmg
Windows: https://desktop-stage.docker.com/win/main/amd64/75249/Docker%20Desktop%20Installer.exe

**Comment by timfallmk** [comment]
> Mac M1: [desktop-stage.docker.com/mac/main/arm64/75234/Docker.dmg](https://desktop-stage.docker.com/mac/main/arm64/75234/Docker.dmg)

Tested and confirmed it will _launch_ backgrounded with the Dock icon hidden. Still getting weird behavior from the cmd + Q key combo when the Dashboard is foregrounded. It seems to "stutter", almost like there are multiple instances of it running.

Quit is still missing from the App menu.
<img width="137" alt="image" src="https://user-images.githubusercontent.com/5316117/156449056-515a07b6-86ae-458d-addf-296b1425fa96.png">

**Comment by mdeliadi** [comment]
I can confirm the background behavior should be fixed, even after a restart, I'm still experiencing the same issues reported by @jirsbek and @timfallmk !

**Comment by trungutt** [comment]
@jirsbek @mdeliadi That sounds something flaky on the Whalemenu... Could you try these builds to see if it could improve the situation? 
Mac Intel: https://desktop-stage.docker.com/mac/main/amd64/75357/Docker.dmg
Mac M1: https://desktop-stage.docker.com/mac/main/arm64/75357/Docker.dmg

**Comment by jirsbek** [comment]
Build 75357 works well. Whalemenu is complete 👍

**Comment by mdeliadi** [comment]
Build 75357 works well also for me. Thank You @trungutt !

**Comment by tonglil** [comment]
Build 75357 on Intel fixed the broken dock/cmd+tab issue for me.

**Comment by mat007** [comment]
We have released Docker Desktop 4.6.0, with a fix for this issue. See https://docs.docker.com/docker-for-mac/release-notes/.
